jetty-websocket/websocket-core/src/test/java/org/eclipse/jetty/websocket/core/server/WebSocketServerTest.java

```diff
@@ -207,29 +204,6 @@ public void onFrame(Frame frame, Callback callback)
             }
             assertThat(serverHandler.receivedFrames.size(), is(5));
             assertThat(receivedCallbacks.size(), is(5));
-
-            byte[] first = serverHandler.receivedFrames.poll().getPayload().array();
-            assertThat(serverHandler.receivedFrames.poll().getPayload().array(), sameInstance(first));
-            assertThat(serverHandler.receivedFrames.poll().getPayload().array(), sameInstance(first));
-            byte[] second = serverHandler.receivedFrames.poll().getPayload().array();
-            assertThat(serverHandler.receivedFrames.poll().getPayload().array(), sameInstance(second));
-            assertThat(first, not(sameInstance(second)));
-
-            ByteBufferPool pool = server.getServer().getConnectors()[0].getByteBufferPool();
-
-            assertThat(pool.acquire(first.length, false).array(), not(sameInstance(first)));
-            receivedCallbacks.poll().succeeded();
-            assertThat(pool.acquire(first.length, false).array(), not(sameInstance(first)));
-            receivedCallbacks.poll().succeeded();
-            assertThat(pool.acquire(first.length, false).array(), not(sameInstance(first)));
-            receivedCallbacks.poll().succeeded();
-            assertThat(pool.acquire(first.length, false).array(), sameInstance(first));
-
-            assertThat(pool.acquire(second.length, false).array(), not(sameInstance(second)));
-            receivedCallbacks.poll().succeeded();
-            assertThat(pool.acquire(second.length, false).array(), not(sameInstance(second)));
-            receivedCallbacks.poll().succeeded();
-            assertThat(pool.acquire(second.length, false).array(), sameInstance(second));
```

Comment:
What happened to these tests?

Reply:
These tests were testing for reference equality on the array wrapped by heap `ByteBuffer`, with the intent to test that the network buffer used to read the websocket frames was actually the same :open_mouth: 
That was super-obscure and super-brittle, as the TCP write could have been split, and they are of course broken if you don't use heap `ByteBuffer`.

Reply:
Ah, yeah. good point.
Resolving.